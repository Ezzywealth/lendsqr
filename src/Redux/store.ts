import { configureStore } from '@reduxjs/toolkit';
import counterReducer from './slices/userSlice';
import { useDispatch } from 'react-redux';

const store = configureStore({
	reducer: {
		counter: counterReducer,
	},
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
export default store;
