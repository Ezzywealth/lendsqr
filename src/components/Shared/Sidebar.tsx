import React from 'react';
import { sidebarObject } from '../../utils/sidebar';
import '../../styles/sidebar.scss';
import { Link } from 'react-router-dom';

const Sidebar = () => {
	const [activeLink, setActiveLink] = React.useState('users');
	return (
		<aside className='sidebar_container'>
			<div className='switch_container'>
				<img src='/assets/icons/switch_organization.png' alt='switch_icon' />
				<h2 className='switch_text'>Switch Organization</h2>
				<img src='/assets/icons/arrow_down.png' alt='switch' />
			</div>
			<div className='dashboard_container'>
				<img src='/assets/icons/dashboard.png' alt='dashboard_icon' />
				<h3 className='dashboard_text'>Dashboard</h3>
			</div>
			<article className='sidebar_links'>
				{Object.entries(sidebarObject).map(([key, value]) => (
					<section key={key} className='links_section'>
						<h3 className='section_title'>{key}</h3>
						<ul className='side_links_container'>
							{value.map((item) => (
								<Link to={item.name === 'users' ? `/${item.name}` : ''} key={item.id}>
									<li className={`side_links ${activeLink === item.name ? 'active_link' : 'inactive_link'}`} onClick={() => setActiveLink(item.name)}>
										<img className='side_links_icon' src={item.icon} alt={item.name} />
										<h4 className='side_links_name'>{item.name}</h4>
									</li>
								</Link>
							))}
						</ul>
					</section>
				))}
			</article>
		</aside>
	);
};

export default Sidebar;
