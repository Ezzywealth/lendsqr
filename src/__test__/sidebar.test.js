/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Sidebar from '../components/Shared/Sidebar';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
// Mock the react-router-dom Link component
jest.mock('react-router-dom', () => ({
	Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

describe('Sidebar component', () => {
	test('renders Pagination component', () => {
		const { container } = render(
			<Provider store={store}>
				<Sidebar />
			</Provider>
		);
		expect(container).toMatchSnapshot();
	});

	test('renders sidebar with links', () => {
		render(
			<Provider store={store}>
				<Sidebar />
			</Provider>
		);
		const dashboardLink = screen.getByText('Dashboard');
		const usersLink = screen.getByText('users');
		expect(dashboardLink).toBeInTheDocument();
		expect(usersLink).toBeInTheDocument();
	});

	test('activates link on click', () => {
		render(
			<Provider store={store}>
				<Sidebar />
			</Provider>
		);

		const usersLink = screen.getByText('users');

		// Click on the Users link
		fireEvent.click(usersLink);

		// Get the immediate parent li element
		const parentLi = usersLink.parentElement;

		// Assert that the parent li element has the 'active_link' class
		expect(parentLi).toHaveClass('active_link');
	});
});
