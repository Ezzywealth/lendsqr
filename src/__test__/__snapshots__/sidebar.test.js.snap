// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders Pagination component 1`] = `
<div>
  <aside
    class="sidebar_container"
  >
    <div
      class="switch_container"
    >
      <img
        alt="switch_icon"
        src="/assets/icons/switch_organization.png"
      />
      <h2
        class="switch_text"
      >
        Switch Organization
      </h2>
      <img
        alt="switch"
        src="/assets/icons/arrow_down.png"
      />
    </div>
    <div
      class="dashboard_container"
    >
      <img
        alt="dashboard_icon"
        src="/assets/icons/dashboard.png"
      />
      <h3
        class="dashboard_text"
      >
        Dashboard
      </h3>
    </div>
    <article
      class="sidebar_links"
    >
      <section
        class="links_section"
      >
        <h3
          class="section_title"
        >
          customers
        </h3>
        <ul
          class="side_links_container"
        >
          <a
            href="/users"
          >
            <li
              class="side_links active_link"
            >
              <img
                alt="users"
                class="side_links_icon"
                src="/assets/icons/users.png"
              />
              <h4
                class="side_links_name"
              >
                users
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="guarantors"
                class="side_links_icon"
                src="/assets/icons/guarantors.png"
              />
              <h4
                class="side_links_name"
              >
                guarantors
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="loans"
                class="side_links_icon"
                src="/assets/icons/loans.png"
              />
              <h4
                class="side_links_name"
              >
                loans
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="decision models"
                class="side_links_icon"
                src="/assets/icons/decision_models.png"
              />
              <h4
                class="side_links_name"
              >
                decision models
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="savings"
                class="side_links_icon"
                src="/assets/icons/savings.png"
              />
              <h4
                class="side_links_name"
              >
                savings
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="loan requests"
                class="side_links_icon"
                src="/assets/icons/loan_request.png"
              />
              <h4
                class="side_links_name"
              >
                loan requests
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="whitelist"
                class="side_links_icon"
                src="/assets/icons/whitelist.png"
              />
              <h4
                class="side_links_name"
              >
                whitelist
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="karma"
                class="side_links_icon"
                src="/assets/icons/karma.png"
              />
              <h4
                class="side_links_name"
              >
                karma
              </h4>
            </li>
          </a>
        </ul>
      </section>
      <section
        class="links_section"
      >
        <h3
          class="section_title"
        >
          businesses
        </h3>
        <ul
          class="side_links_container"
        >
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="organization"
                class="side_links_icon"
                src="/assets/icons/organization.png"
              />
              <h4
                class="side_links_name"
              >
                organization
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="loan products"
                class="side_links_icon"
                src="/assets/icons/loan_product.png"
              />
              <h4
                class="side_links_name"
              >
                loan products
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="savings products"
                class="side_links_icon"
                src="/assets/icons/savings_product.png"
              />
              <h4
                class="side_links_name"
              >
                savings products
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="fees and charges"
                class="side_links_icon"
                src="/assets/icons/fees_charges.png"
              />
              <h4
                class="side_links_name"
              >
                fees and charges
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="transactions"
                class="side_links_icon"
                src="/assets/icons/transactions.png"
              />
              <h4
                class="side_links_name"
              >
                transactions
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="services"
                class="side_links_icon"
                src="/assets/icons/services.png"
              />
              <h4
                class="side_links_name"
              >
                services
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="service account"
                class="side_links_icon"
                src="/assets/icons/service_account.png"
              />
              <h4
                class="side_links_name"
              >
                service account
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="settlements"
                class="side_links_icon"
                src="/assets/icons/settlements.png"
              />
              <h4
                class="side_links_name"
              >
                settlements
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="reports"
                class="side_links_icon"
                src="/assets/icons/reports.png"
              />
              <h4
                class="side_links_name"
              >
                reports
              </h4>
            </li>
          </a>
        </ul>
      </section>
      <section
        class="links_section"
      >
        <h3
          class="section_title"
        >
          settings
        </h3>
        <ul
          class="side_links_container"
        >
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="preferences"
                class="side_links_icon"
                src="/assets/icons/preferences.png"
              />
              <h4
                class="side_links_name"
              >
                preferences
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="fees and pricing"
                class="side_links_icon"
                src="/assets/icons/fees_pricing.png"
              />
              <h4
                class="side_links_name"
              >
                fees and pricing
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="audit logs"
                class="side_links_icon"
                src="/assets/icons/audits_logs.png"
              />
              <h4
                class="side_links_name"
              >
                audit logs
              </h4>
            </li>
          </a>
          <a
            href=""
          >
            <li
              class="side_links inactive_link"
            >
              <img
                alt="Systems Messages"
                class="side_links_icon"
                src="/assets/icons/system_messages.png"
              />
              <h4
                class="side_links_name"
              >
                Systems Messages
              </h4>
            </li>
          </a>
        </ul>
      </section>
      <div
        class="logout_container"
      >
        <img
          alt="logout button"
          class="side_links_icon"
          src="/assets/icons/logout.png"
        />
        <h3
          class="logout_text"
        >
          Logout
        </h3>
      </div>
    </article>
  </aside>
</div>
`;
